### PR TITLE
Temp PoC button to create Cloud Task from UI (with proper auth)

### PR DESCRIPTION
**Pull request by johanehr**
This PR adds a temporary page with a UI button to trigger an immediate hardcoded email via Google Cloud Tasks. As clean up (sorry for the large diff!), it also removes the gcp-backend folder, which included previous minimal tests of the used GCP functionality.

As a follow up, this can be adapted to work with the planned frontend changes, but will naturally require feeding in more user input. Additionally, changes will have to be made so that any scheduled task with the same name is replaced (for when a vehicle is moved before a reminder is set off). Both of these are future changes though.

Some useful links regarding proper auth with a local or cloud deployment:
https://cloud.google.com/tasks/docs/samples/cloud-tasks-create-http-task-with-token
https://cloud.google.com/tasks/docs/creating-http-target-tasks#sa
https://cloud.google.com/docs/authentication/application-default-credentials#attached-sa

### CODE REVIEW

File: .env.local.example [hunk 1]
```diff
@@ -1,2 +1,10 @@
 NEXT_PUBLIC_GOOGLE_MAPS_API_KEY=CREATE_YOUR_OWN_KEY
-NEXT_PUBLIC_GOOGLE_MAPS_MAP_ID=NUMBERS_AND_STUFF
\ No newline at end of file
+NEXT_PUBLIC_GOOGLE_MAPS_MAP_ID=NUMBERS_AND_STUFF
+
+# Google Cloud stuff for backend functionality
```
Comment (johanehr) [inline .env.local.example:4]: Make sure to update your .env.local with these! I can help you set up the Base64 key later.

File: pages/api/trigger-reminder.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    location: {
+      name: 'Storgatan 12',
+      lat: 59.330741,
+      lng: 18.032743
+    },
+    move_by_timestamp: '2024-05-02T16:10:18Z'
+  }
+
+  const name = client.taskPath(project, location, queue, `EMAIL_luhcforghgmailcom_${Date.now()}`)
+  // TODO: Make sure this overwrites any existing tasks with the same name (use method, email, and nickname as input)
```
Comment (johanehr) [inline pages/api/trigger-reminder.ts:45]: This is planned to be handled in a follow-up PR

File: pages/api/trigger-reminder.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+auth.fromJSON(credential)
+const client = new CloudTasksClient({
+  auth: auth
+})
+
+// Based on https://cloud.google.com/tasks/docs/creating-http-target-tasks#advanced_task_creation_createtask_method
+async function createHttpTask(nickname: string = 'DEFAULT') {
+  const parent = client.queuePath(project, location, queue)
+
+  // TODO: Modify this function to accept input for the following fields (as accepted by the current email template)
```
Comment (johanehr) [inline pages/api/trigger-reminder.ts:32]: As a next step, once this and https://github.com/johanehr/parking-reminder-frontend/pull/32 are merged, we can tie them together. As such, this TODO is left until a later PR.

**Review by frankosgit — APPROVED** [review]
looks great from what I can see, good work! :)

**Pull request merged**